lib/router/src/cache.rs

```diff
@@ -144,21 +141,8 @@ where
     type Error = ();
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        loop {
-            try_ready!(self.interval.poll().map_err(|e| {
-                panic!("Interval::poll must not fail: {}", e);
-            }));
-            debug!("purge task interval reached");
-
-            let mut cache = try_ready!(Ok(self.cache.poll_lock()));
-            debug!("purge task locked the router cache");
-
-            // If poll_purge is not ready, we cannot expire any values and
-            // wait for the next interval. If poll_purge is ready, we have
-            // expired all values and wait for the next interval
-            debug!("purge task is polling for evictions...");
-            cache.poll_purge().expect("Cache::poll_purge must not fail");
-        }
+        let mut cache = try_ready!(Ok(self.0.poll_lock()));
+        cache.poll_purge()
```

Comment:
Nit/TIOLI: does
```rust
try_ready!(Ok(self.0.poll_lock())).poll_purge()
```
not work, did you think this was clearer?

Reply:
it works, but there are a lot of parens there, so i thought it clearer...